File: client-module-server/src/main/java/br/edu/les/module/client/domain/Cidade.java
package br.edu.les.module.client.domain;

import lombok.*;
import org.springframework.stereotype.Component;

import javax.persistence.*;

@Builder
@Component
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "cidade")
public class Cidade extends EntidadeDominio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cid_id")
    private Integer id;

    @Column(name = "cid_descricao")
    private String descricao;

    @ManyToOne
    @JoinColumn(name = "cid_estado_id")
    private Estado estado;
}

File: client-module-server-node/src/models/Cidade.ts
import { Column, Entity, JoinColumn, ManyToOne, OneToMany, PrimaryGeneratedColumn } from "typeorm";
import { Endereco } from "./Endereco";
import { Estado } from "./Estado";

@Entity("cidade")
class Cidade {
    
    @PrimaryGeneratedColumn({name: "cid_id"})
    id: number;

    @Column({name: "cid_descricao"})
    descricao: string;

    @ManyToOne(type => Estado, estado => estado.cidades, {eager: true})
    @JoinColumn({name: "cid_estado_id"})
    estado: Estado;

    @OneToMany(type => Endereco, enderecos => enderecos.cidade)
    enderecos: Endereco[];
}

export { Cidade }